x/plans/keeper/migrations.go

```diff
@@ -22,3 +25,39 @@ func (m Migrator) Migrate2to3(ctx sdk.Context) error {
 	}
 	return nil
 }
+
+func (m Migrator) Migrate3to4(ctx sdk.Context) error {
+	planIndices := m.keeper.GetAllPlanIndices(ctx)
+
+	for _, planIndex := range planIndices {
```

Comment:
This would update only the latest version of each plan, and not older fixated versions that may still be in use. You need to use fixation's GetAllEntryVersions(ctx, entry, true) to get the list of block (versions) for each index, and (nested) loop on them.